snap/info.go

```diff
@@ -183,8 +183,10 @@ type Info struct {
 	Prices  map[string]float64
 	MustBuy bool
 
-	PublisherID string
-	Publisher   string
+	// Publisher's ID, username, and display-name
+	PublisherID          string
+	Publisher            string
+	PublisherDisplayName string
```

Comment:
why are we flattening it in `snap.Info`? We might as well make it structured at this point. Or am I missing something?

Reply:
smaller diff, do you feel strongly about that in here?

Reply:
I could move to share snap.StoreAccount

Reply:
I have done this, and indeed the changes were a bit large but not PR-explodingly so